File: travelai/backend/app/Models/Place.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Place extends Model
{
    use HasFactory;

    protected $fillable = [
        'category_id',
        'name',
        'description',
        'address',
        'latitude',
        'longitude',
        'image_url',
        'rating_avg',
    ];

    protected $casts = [
        'latitude' => 'float',
        'longitude' => 'float',
        'rating_avg' => 'float',
    ];

    public $timestamps = false;

    public function category()
    {
        return $this->belongsTo(Category::class);
    }

    public function reviews()
    {
        return $this->hasMany(Review::class);
    }

    public function favoritedByUsers()
    {
        return $this->belongsToMany(User::class, 'favorites', 'place_id', 'user_id');
    }
}

File: travelai/travelai_flutter/lib/models/place.dart
class Place {
  final int id;
  final int categoryId;
  final String name;
  final String description;
  final String address;
  final double latitude;
  final double longitude;
  final String imageUrl;
  double ratingAvg;

  Place({
    required this.id,
    required this.categoryId,
    required this.name,
    required this.description,
    required this.address,
    required this.latitude,
    required this.longitude,
    required this.imageUrl,
    required this.ratingAvg,
  });

  Map<String, dynamic> toMap() => {
        'id': id,
        'categoryId': categoryId,
        'name': name,
        'description': description,
        'address': address,
        'latitude': latitude,
        'longitude': longitude,
        'imageUrl': imageUrl,
        'ratingAvg': ratingAvg,
      };

  factory Place.fromMap(Map<String, dynamic> map) => Place(
        id: map['id'] as int,
        categoryId: map['categoryId'] as int,
        name: map['name'] as String,
        description: map['description'] as String,
        address: map['address'] as String,
        latitude: (map['latitude'] as num).toDouble(),
        longitude: (map['longitude'] as num).toDouble(),
        imageUrl: map['imageUrl'] as String,
        ratingAvg: (map['ratingAvg'] as num).toDouble(),
      );
}
